### ISSUE TITLE
Harmony 4.0 not supported

### ISSUE THREAD

**Issue by dbsxdbsx** [chris-wolf/autostart_settings#2]
For Harmoney 4.0, the setting page can't be triggered out:
```dart
import 'dart:io';

import 'package:autostart_settings/autostart_settings.dart';
import 'package:flutter/material.dart';
import 'package:launch_at_startup/launch_at_startup.dart';
import 'package:package_info_plus/package_info_plus.dart';
import 'package:proxy/common/utils.dart';

class AutoStart {
  static Future<bool> setAutoStart(bool enable) async {
    if (isDesktop()) {
      WidgetsFlutterBinding.ensureInitialized();
      final packageInfo = await PackageInfo.fromPlatform();
      launchAtStartup.setup(
        appName: packageInfo.appName,
        appPath: Platform.resolvedExecutable,
        packageName: 'dev.leanflutter.examples.launchatstartupexample',
      );

      if (enable) {
        return launchAtStartup.enable();
      } else {
        return launchAtStartup.disable();
      }
    }
    // mobile
    if (Platform.isAndroid) {
      // final flutterAutostart = FlutterAutostart();
      final canOpen =
          // await AutostartSettings.canOpen(autoStart: true, batterySafer: true);
          await AutostartSettings.canOpen(autoStart: true, batterySafer: false);
      if (canOpen) {
        return await AutostartSettings.open(
            autoStart: true, batterySafer: true);
      } else {
        return false; // reaching here
      }
    }
    return false;
  }
}
```
it would reach here: `return false; // reaching here`.

**Comment by chris-wolf** [comment]
Autostart settings only exists for older Huawei devices as far as I know that are opened with:
Pair(
                "com.huawei.systemmanager",
                "com.huawei.systemmanager.optimize.process.ProtectActivity"
            )

For Harmony 4.0 I couldn't find any info on the internet if autostart settings exists. So autostart probably should work out of the box if it is configures in the AndroidManifest.xml correctly.

If there exists an autostart settings page that need to granted I would be glad to add it to the list of autostart activity intents. :)
You could extract the necessary info via adb:
-  list all activityName:  adb shell dumpsys package | grep Activity
-  finding current active/resumed activity: adb shell dumpsys activity activities | grep mResumedActivity

**Comment by dbsxdbsx** [comment]
When the mobile is at the program auto start page, with this cmd `db shell dumpsys activity activities | findstr mResumedActivity`, adb feedback:
```
mResumedActivity: ActivityRecord{4c6d1af u0 com.huawei.systemmanager/.appcontrol.activity.StartupAppControlActivity t25}
```

**Comment by chris-wolf** [comment]
Awesome thanks!
I released version 0.1.4 to include this intent.

Unfortunately it seems that with a security update from 2021 Huawei prevents other app from launching the settings so this might not work anymore. 😕

> the intent for "App Launch" settings is not supported anymore on all EMUI since beginning of 2021 due to security reasons.

https://stackoverflow.com/a/70801185/13404130

But couldn't really find much details about this.

**Comment by dbsxdbsx** [comment]
Now with v0.1.4, the `canOpen` var in my code returns `true`, but it can't navigate to the page.  Maybe it is due to the security issue as you said. 
Even if it is manually set to auto-start, it doesn't auto-start.   
So I think the issue is mainly due to Harmony.

Anyway, thanks!
